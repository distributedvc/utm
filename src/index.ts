import { parse, pick } from 'query-string';

type UtmParams = {
  utm_source?: string;
  utm_medium?: string;
  utm_term?: string;
  utm_content?: string;
  utm_name?: string;
  [key: string]: string | string[] | undefined | null | boolean | number;
};

export function utm(query: string = window?.location?.search): UtmParams {
  const utms = pick(query, (name) => name.startsWith('utm_'));
  return parse(utms) as UtmParams;
}
