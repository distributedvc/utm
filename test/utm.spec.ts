import { utm } from '../src';

describe('UTM', (): void => {
  it('should extract correct utms from the query', async (): Promise<void> => {
    const utms = utm(
      '?utm_source=google&utm_campaign=name&utm_medium=medium&utm_content=content&utm_term=term'
    );

    expect(utms).toMatchSnapshot();
  });

  it('should ignore non utms params', async (): Promise<void> => {
    const utms = utm(
      '?utm_source=google&utm_campaign=name&utm_medium=medium&utm_content=content&utm_term=term&foo=bar&baz=foo'
    );

    expect(utms).toMatchSnapshot();
  });
});
