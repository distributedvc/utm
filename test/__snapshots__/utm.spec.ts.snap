// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`UTM should extract correct utms from the query 1`] = `
Object {
  "utm_campaign": "name",
  "utm_content": "content",
  "utm_medium": "medium",
  "utm_source": "google",
  "utm_term": "term",
}
`;

exports[`UTM should ignore non utms params 1`] = `
Object {
  "utm_campaign": "name",
  "utm_content": "content",
  "utm_medium": "medium",
  "utm_source": "google",
  "utm_term": "term",
}
`;
